twilight-model/src/channel/message/component/select_menu.rs

```diff
@@ -1,27 +1,59 @@
-use crate::channel::message::ReactionType;
+use crate::channel::{message::ReactionType, ChannelType};
 use serde::{Deserialize, Serialize};
 
-/// Dropdown-style [`Component`] that renders belew messages.
+/// Dropdown-style [`Component`] that renders below messages.
 ///
 /// [`Component`]: super::Component
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct SelectMenu {
+    /// An optional list of channel types.
+    ///
+    /// This option is only used for [channel select menus](SelectMenuType::Channel).
+    pub channel_types: Option<Vec<ChannelType>>,
     /// Developer defined identifier.
     pub custom_id: String,
     /// Whether the select menu is disabled.
     ///
     /// Defaults to `false`.
     pub disabled: bool,
+    /// This select menu's type.
+    ///
+    /// [Text select menus](SelectMenuType::Text) *must* also set the `options` field.
+    pub kind: SelectMenuType,
     /// Maximum number of options that may be chosen.
     pub max_values: Option<u8>,
     /// Minimum number of options that must be chosen.
     pub min_values: Option<u8>,
-    /// List of available choices.
-    pub options: Vec<SelectMenuOption>,
+    /// A list of available options.
+    ///
+    /// This value is only used and required by [text select menus](SelectMenuType::Text).
+    pub options: Option<Vec<SelectMenuOption>>,
     /// Custom placeholder text if no option is selected.
     pub placeholder: Option<String>,
 }
 
+/// A [`SelectMenu`]'s type.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[non_exhaustive]
+pub enum SelectMenuType {
+    /// Select menus with a text-based `options` list.
+    ///
+    /// Select menus of this `kind` *must* set the `options` field to specify the options users
```

Comment:
I don't think we should be writing API documentation for Discord in all places imaginable, one should be enough on the `options` field

Reply:
I mention this in the enum documentation, as users may only read its documentation to see which select menu types are available.
Now that the `options` field is an `Option<_>`, I think the duplicated documentation is worth it if it saves some people from debugging a Discord API error.
However, I'm open to talking about this instance.

Reply:
any other opinions on this or do we leave it as is?

Reply:
I agree upon that this duplicated documentation is worth it; in terms of helping users understand the requirements on fields for specific select menu types.

As such, I believe that we can keep this as is.